src/miner_lora.erl

```diff
@@ -151,9 +155,14 @@ mk_token(Timers) ->
         false -> Token
     end.
 
--spec select_gateway(map()) -> gateway().
+-spec select_gateway(map()) -> {ok, gateway()} | {error, no_gateways}.
 select_gateway(Gateways) ->
-    erlang:element(2, erlang:hd(maps:to_list(Gateways))).
+    case maps:size(Gateways) of
+        0 ->
+            {error, no_gateways};
+        _ ->
+            {ok, erlang:element(2, erlang:hd(maps:to_list(Gateways)))}
```

Comment:
Select the first gateway always? Can we add a comment here if we need to update this down the line?

Reply:
yeah, this will change with the 'multi tenant' miner

Reply:
Added a comment